lib/src/Status/Monitoring/Monitor.cpp

```diff
@@ -28,18 +28,59 @@ void Monitor::initialize(QQmlApplicationEngine* engine) {
     }, Qt::QueuedConnection);
 }
 
-QStringList Monitor::getContextPropertiesNames() const
+ContextPropertiesModel* Monitor::contexPropertiesModel()
 {
-    return m_contexPropertiesNames;
+    return &m_contexPropertiesModel;
 }
 
 void Monitor::addContextPropertyName(const QString &contextPropertyName)
 {
-    if (m_contexPropertiesNames.contains(contextPropertyName))
-        return;
+    m_contexPropertiesModel.addContextProperty(contextPropertyName);
+}
+
+bool Monitor::isModel(QVariant obj) const
+{
+    if (!obj.canConvert<QObject*>())
+        return false;
+
+    return qobject_cast<QAbstractItemModel*>(obj.value<QObject*>()) != nullptr;
+}
+
+QString Monitor::typeName(QVariant obj) const
+{
+    if (obj.canConvert<QObject*>())
+        return obj.value<QObject*>()->metaObject()->className();
+
+    return QString::fromUtf8(obj.typeName());
+}
+
+QJSValue Monitor::modelRoles(QAbstractItemModel *model) const
+{
+    if (model == nullptr)
+        return {};
+
+    QJSEngine *engine = qjsEngine(this);
+
+    if (engine == nullptr)
+        return {};
+
+    const auto& roleNames = model->roleNames();
+
+    QJSValue array = engine->newArray(roleNames.size());
+    QList<int> keys = roleNames.keys();
+
+    for (auto i = 0; i < keys.size(); i++) {
+        QJSValue item = engine->newObject();
```

Comment:
Oh and, careful about the ownership here:

```
QJSValue QJSEngine::newQObject(QObject *object) Creates a JavaScript object that wraps the given QObject object, using JavaScriptOwnership.
```

Reply:
But do you see any problems here? I can't see any danger here, final object doesn't depend on the input object, it's independent.